packages/cli-repl/src/cli-repl.ts

```diff
@@ -152,52 +151,16 @@ class CliRepl {
     });
 
     this.repl.on('exit', () => {
-      this.serviceProvider.close(true);
+      this.internalState.close(true);
```

Comment:
This one was the same before, but i just noticed it: this function without await can both throw or reject.

Another thing to be noted. If we don't await  `close`, `process.exit()` will likely kill the process before anything would happen about closing the service provider gracefully.

I don't see a practical downside here, just something to keep in mind.

Reply:
I've updated it so it will await each `close` so that we get through everything